shell/platform/darwin/ios/framework/Source/FlutterOverlayView.h

```diff
@@ -23,6 +23,7 @@
 - (instancetype)initWithCoder:(NSCoder*)aDecoder NS_UNAVAILABLE;
 
 - (instancetype)init NS_DESIGNATED_INITIALIZER;
+- (instancetype)initForGLWithContentsScale:(CGFloat)contentsScale;
```

Comment:
Just say `initWithContentsScale` and check the layer class in the implementation before setting the content and rasterization scales.